static/app/components/charts/worldMapChart.tsx

```diff
@@ -46,136 +38,121 @@ const DEFAULT_CENTER_X = 10.97;
 const DISCOVER_QUERY_LIST_CENTER_Y = -12;
 const DEFAULT_CENTER_Y = 9.71;
 
-class WorldMapChart extends React.Component<Props, State> {
-  state: State = {
-    countryToCodeMap: null,
-    map: null,
-    codeToCountryMap: null,
-  };
+export function WorldMapChart({
+  series,
+  seriesOptions,
+  fromDiscover,
+  fromDiscoverQueryList,
+  ...props
+}: WorldMapChartProps) {
+  const theme = useTheme();
 
-  async componentDidMount() {
-    const [countryToCodeMap, worldMap] = await Promise.all([
-      import('sentry/data/countryCodesMap'),
-      import('sentry/data/world.json'),
-    ]);
-
-    // Echarts not available in tests
-    echarts.registerMap?.('sentryWorld', worldMap.default as any);
-
-    // eslint-disable-next-line
-    this.setState({
-      countryToCodeMap: countryToCodeMap.default,
-      map: worldMap.default,
-      codeToCountryMap: Object.fromEntries(
-        Object.entries(countryToCodeMap.default).map(([country, code]) => [code, country])
-      ),
-    });
-  }
-
-  render() {
-    const {countryToCodeMap, map} = this.state;
-
-    if (countryToCodeMap === null || map === null) {
-      return null;
+  const codeToCountryMap = useMemo(() => {
+    const countryMaps: Record<string, string> = {};
+
+    for (const country in worldMap) {
+      countryMaps[countryToCodeMap[country]] = country;
     }
 
-    const {series, seriesOptions, theme, fromDiscover, fromDiscoverQueryList, ...props} =
-      this.props;
-    const processedSeries = series.map(({seriesName, data, ...options}) =>
-      MapSeries({
-        ...seriesOptions,
-        ...options,
-        map: 'sentryWorld',
-        name: seriesName,
-        nameMap: this.state.countryToCodeMap ?? undefined,
-        aspectScale: 0.85,
-        zoom: fromDiscover
-          ? DISCOVER_ZOOM
-          : fromDiscoverQueryList
-          ? DISCOVER_QUERY_LIST_ZOOM
-          : DEFAULT_ZOOM,
-        center: [
-          DEFAULT_CENTER_X,
-          fromDiscoverQueryList ? DISCOVER_QUERY_LIST_CENTER_Y : DEFAULT_CENTER_Y,
-        ],
+    return countryMaps;
+  }, []);
+
+  useEffect(() => {
+    echarts.registerMap?.('sentryWorld', worldMap as any);
```

Comment:
why the as any cast? what happens if you just remove that

Reply:
we get a type error. maybe we can fix it in a follow-up PR